python/setup.py.in

```diff
@@ -44,7 +44,11 @@ def get_hdk_shared_libs() -> list:
 
 
 def gen_compile_flags() -> list:
-    cpp_version_flags = ["-std=c++17"]
+    cpp_version_flags = [
+        "-std=c++17",
+        "-Wno-unused-variable",
```

Comment:
Are any of these warnings real?

Reply:
I am getting this errors during build. 
-- The C compiler identification is GNU 12.2.0
-- The CXX compiler identification is GNU 12.2.0

Do you have such warns? 
```
/localdisk/dmitriim/hdk/python/pyhdk/_execute.cpp:4288:42: warning: '__pyx_f_7pyarrow_3lib_pyarrow_unwrap_table' defined but not used [-Wunused-variable]
 4288 | static std::shared_ptr< arrow::Table>  (*__pyx_f_7pyarrow_3lib_pyarrow_unwrap_table)(PyObject *); /*proto*/
      |                                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/localdisk/dmitriim/hdk/python/pyhdk/_execute.cpp:4287:48: warning: '__pyx_f_7pyarrow_3lib_pyarrow_unwrap_batch' defined but not used [-Wunused-variable]
 4287 | static std::shared_ptr< arrow::RecordBatch>  (*__pyx_f_7pyarrow_3lib_pyarrow_unwrap_batch)(PyObject *); /*proto*/
      |                                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/localdisk/dmitriim/hdk/python/pyhdk/_execute.cpp:4286:43: warning: '__pyx_f_7pyarrow_3lib_pyarrow_unwrap_tensor' defined but not used [-Wunused-variable]
 4286 | static std::shared_ptr< arrow::Tensor>  (*__pyx_f_7pyarrow_3lib_pyarrow_unwrap_tensor)(PyObject *); /*proto*/
      |                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/localdisk/dmitriim/hdk/python/pyhdk/_execute.cpp:4285:52: warning: '__pyx_f_7pyarrow_3lib_pyarrow_unwrap_sparse_csr_matrix' defined but not used [-Wunused-variable]
 4285 | static std::shared_ptr< arrow::SparseCSRMatrix>  (*__pyx_f_7pyarrow_3lib_pyarrow_unwrap_sparse_csr_matrix)(PyObject *); /*proto*/
      |                                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/localdisk/dmitriim/hdk/python/pyhdk/_execute.cpp:4284:52: warning: '__pyx_f_7pyarrow_3lib_pyarrow_unwrap_sparse_csf_tensor' defined but not used [-Wunused-variable]
 4284 | static std::shared_ptr< arrow::SparseCSFTensor>  (*__pyx_f_7pyarrow_3lib_pyarrow_unwrap_sparse_csf_tensor)(PyObject *); /*proto*/
      |                                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/localdisk/dmitriim/hdk/python/pyhdk/_execute.cpp:4283:52: warning: '__pyx_f_7pyarrow_3lib_pyarrow_unwrap_sparse_csc_matrix' defined but not used [-Wunused-variable]
 4283 | static std::shared_ptr< arrow::SparseCSCMatrix>  (*__pyx_f_7pyarrow_3lib_pyarrow_unwrap_sparse_csc_matrix)(PyObject *); /*proto*/
      |                                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/localdisk/dmitriim/hdk/python/pyhdk/_execute.cpp:4282:52: warning: '__pyx_f_7pyarrow_3lib_pyarrow_unwrap_sparse_coo_tensor' defined but not used [-Wunused-variable]
 4282 | static std::shared_ptr< arrow::SparseCOOTensor>  (*__pyx_f_7pyarrow_3lib_pyarrow_unwrap_sparse_coo_tensor)(PyObject *); /*proto*/
      |                                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/localdisk/dmitriim/hdk/python/pyhdk/_execute.cpp:4281:49: warning: '__pyx_f_7pyarrow_3lib_pyarrow_unwrap_chunked_array' defined but not used [-Wunused-variable]
 4281 | static std::shared_ptr< arrow::ChunkedArray>  (*__pyx_f_7pyarrow_3lib_pyarrow_unwrap_chunked_array)(PyObject *); /*proto*/
      |                                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/localdisk/dmitriim/hdk/python/pyhdk/_execute.cpp:4280:42: warning: '__pyx_f_7pyarrow_3lib_pyarrow_unwrap_array' defined but not used [-Wunused-variable]
 4280 | static std::shared_ptr< arrow::Array>  (*__pyx_f_7pyarrow_3lib_pyarrow_unwrap_array)(PyObject *); /*proto*/
      |                                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/localdisk/dmitriim/hdk/python/pyhdk/_execute.cpp:4279:43: warning: '__pyx_f_7pyarrow_3lib_pyarrow_unwrap_scalar' defined but not used [-Wunused-variable]
 4279 | static std::shared_ptr< arrow::Scalar>  (*__pyx_f_7pyarrow_3lib_pyarrow_unwrap_scalar)(PyObject *); /*proto*/
      |                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/localdisk/dmitriim/hdk/python/pyhdk/_execute.cpp:4278:43: warning: '__pyx_f_7pyarrow_3lib_pyarrow_unwrap_schema' defined but not used [-Wunused-variable]
 4278 | static std::shared_ptr< arrow::Schema>  (*__pyx_f_7pyarrow_3lib_pyarrow_unwrap_schema)(PyObject *); /*proto*/
      |                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/localdisk/dmitriim/hdk/python/pyhdk/_execute.cpp:4277:42: warning: '__pyx_f_7pyarrow_3lib_pyarrow_unwrap_field' defined but not used [-Wunused-variable]
 4277 | static std::shared_ptr< arrow::Field>  (*__pyx_f_7pyarrow_3lib_pyarrow_unwrap_field)(PyObject *); /*proto*/
      |                                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/localdisk/dmitriim/hdk/python/pyhdk/_execute.cpp:4276:45: warning: '__pyx_f_7pyarrow_3lib_pyarrow_unwrap_data_type' defined but not used [-Wunused-variable]
 4276 | static std::shared_ptr< arrow::DataType>  (*__pyx_f_7pyarrow_3lib_pyarrow_unwrap_data_type)(PyObject *); /*proto*/
      |                                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/localdisk/dmitriim/hdk/python/pyhdk/_execute.cpp:4275:43: warning: '__pyx_f_7pyarrow_3lib_pyarrow_unwrap_buffer' defined but not used [-Wunused-variable]
 4275 | static std::shared_ptr< arrow::Buffer>  (*__pyx_f_7pyarrow_3lib_pyarrow_unwrap_buffer)(PyObject *); /*proto*/
      |                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/localdisk/dmitriim/hdk/python/pyhdk/_execute.cpp:4274:20: warning: '__pyx_f_7pyarrow_3lib_pyarrow_wrap_table' defined but not used [-Wunused-variable]
 4274 | static PyObject *(*__pyx_f_7pyarrow_3lib_pyarrow_wrap_table)(std::shared_ptr< arrow::Table>  const &); /*proto*/
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/localdisk/dmitriim/hdk/python/pyhdk/_execute.cpp:4273:20: warning: '__pyx_f_7pyarrow_3lib_pyarrow_wrap_batch' defined but not used [-Wunused-variable]
 4273 | static PyObject *(*__pyx_f_7pyarrow_3lib_pyarrow_wrap_batch)(std::shared_ptr< arrow::RecordBatch>  const &); /*proto*/
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/localdisk/dmitriim/hdk/python/pyhdk/_execute.cpp:4272:20: warning: '__pyx_f_7pyarrow_3lib_pyarrow_wrap_tensor' defined but not used [-Wunused-variable]
 4272 | static PyObject *(*__pyx_f_7pyarrow_3lib_pyarrow_wrap_tensor)(std::shared_ptr< arrow::Tensor>  const &); /*proto*/
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/localdisk/dmitriim/hdk/python/pyhdk/_execute.cpp:4271:20: warning: '__pyx_f_7pyarrow_3lib_pyarrow_wrap_sparse_csr_matrix' defined but not used [-Wunused-variable]
 4271 | static PyObject *(*__pyx_f_7pyarrow_3lib_pyarrow_wrap_sparse_csr_matrix)(std::shared_ptr< arrow::SparseCSRMatrix>  const &); /*proto*/
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/localdisk/dmitriim/hdk/python/pyhdk/_execute.cpp:4270:20: warning: '__pyx_f_7pyarrow_3lib_pyarrow_wrap_sparse_csf_tensor' defined but not used [-Wunused-variable]
 4270 | static PyObject *(*__pyx_f_7pyarrow_3lib_pyarrow_wrap_sparse_csf_tensor)(std::shared_ptr< arrow::SparseCSFTensor>  const &); /*proto*/
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/localdisk/dmitriim/hdk/python/pyhdk/_execute.cpp:4269:20: warning: '__pyx_f_7pyarrow_3lib_pyarrow_wrap_sparse_csc_matrix' defined but not used [-Wunused-variable]
 4269 | static PyObject *(*__pyx_f_7pyarrow_3lib_pyarrow_wrap_sparse_csc_matrix)(std::shared_ptr< arrow::SparseCSCMatrix>  const &); /*proto*/
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/localdisk/dmitriim/hdk/python/pyhdk/_execute.cpp:4268:20: warning: '__pyx_f_7pyarrow_3lib_pyarrow_wrap_sparse_coo_tensor' defined but not used [-Wunused-variable]
 4268 | static PyObject *(*__pyx_f_7pyarrow_3lib_pyarrow_wrap_sparse_coo_tensor)(std::shared_ptr< arrow::SparseCOOTensor>  const &); /*proto*/
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/localdisk/dmitriim/hdk/python/pyhdk/_execute.cpp:4267:20: warning: '__pyx_f_7pyarrow_3lib_pyarrow_wrap_chunked_array' defined but not used [-Wunused-variable]
 4267 | static PyObject *(*__pyx_f_7pyarrow_3lib_pyarrow_wrap_chunked_array)(std::shared_ptr< arrow::ChunkedArray>  const &); /*proto*/
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/localdisk/dmitriim/hdk/python/pyhdk/_execute.cpp:4266:20: warning: '__pyx_f_7pyarrow_3lib_pyarrow_wrap_array' defined but not used [-Wunused-variable]
 4266 | static PyObject *(*__pyx_f_7pyarrow_3lib_pyarrow_wrap_array)(std::shared_ptr< arrow::Array>  const &); /*proto*/
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/localdisk/dmitriim/hdk/python/pyhdk/_execute.cpp:4265:20: warning: '__pyx_f_7pyarrow_3lib_pyarrow_wrap_scalar' defined but not used [-Wunused-variable]
 4265 | static PyObject *(*__pyx_f_7pyarrow_3lib_pyarrow_wrap_scalar)(std::shared_ptr< arrow::Scalar>  const &); /*proto*/
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/localdisk/dmitriim/hdk/python/pyhdk/_execute.cpp:4264:20: warning: '__pyx_f_7pyarrow_3lib_pyarrow_wrap_schema' defined but not used [-Wunused-variable]
 4264 | static PyObject *(*__pyx_f_7pyarrow_3lib_pyarrow_wrap_schema)(std::shared_ptr< arrow::Schema>  const &); /*proto*/
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/localdisk/dmitriim/hdk/python/pyhdk/_execute.cpp:4263:20: warning: '__pyx_f_7pyarrow_3lib_pyarrow_wrap_field' defined but not used [-Wunused-variable]
 4263 | static PyObject *(*__pyx_f_7pyarrow_3lib_pyarrow_wrap_field)(std::shared_ptr< arrow::Field>  const &); /*proto*/
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/localdisk/dmitriim/hdk/python/pyhdk/_execute.cpp:4262:20: warning: '__pyx_f_7pyarrow_3lib_pyarrow_wrap_data_type' defined but not used [-Wunused-variable]
 4262 | static PyObject *(*__pyx_f_7pyarrow_3lib_pyarrow_wrap_data_type)(std::shared_ptr< arrow::DataType>  const &); /*proto*/
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/localdisk/dmitriim/hdk/python/pyhdk/_execute.cpp:4261:20: warning: '__pyx_f_7pyarrow_3lib_pyarrow_wrap_resizable_buffer' defined but not used [-Wunused-variable]
 4261 | static PyObject *(*__pyx_f_7pyarrow_3lib_pyarrow_wrap_resizable_buffer)(std::shared_ptr< arrow::ResizableBuffer>  const &); /*proto*/
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/localdisk/dmitriim/hdk/python/pyhdk/_execute.cpp:4260:20: warning: '__pyx_f_7pyarrow_3lib_pyarrow_wrap_buffer' defined but not used [-Wunused-variable]
 4260 | static PyObject *(*__pyx_f_7pyarrow_3lib_pyarrow_wrap_buffer)(std::shared_ptr< arrow::Buffer>  const &); /*proto*/
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /localdisk/dmitriim/hdk/python/pyhdk/_execute.cpp:911:
/localdisk/dmitriim/miniconda3/envs/omnisci-dev/include/python3.9/datetime.h:189:25: warning: 'PyDateTimeAPI' defined but not used [-Wunused-variable]
  189 | static PyDateTime_CAPI *PyDateTimeAPI = NULL;
      |                         ^~~~~~~~~~~~~
```

Reply:
I am less concerned about maybe unused (especially since so many are in 3rd party libs) but what about maybe uninitialized? That could be a real warning unless it's all in third party stuff.

Reply:
Another possible solution is add maybe unused in pyx. Is it better way?

Reply:
I don't think there are some unused variable warnings relevant to our code. And in this warnings flow it would be hard to see unused variables related to our code anyway, so I'm OK to disable these warnings.

The `maybe-uninitialized` one might be relevant, so I would double-check it.